src/preprocessors/timeseries.test.js

```diff
@@ -1,10 +1,24 @@
-import { createFieldName, generateSortedTimestamps, generateTimestampMap, DataSource } from './datasource.ts';
 import { setTemplateSrv } from '@grafana/runtime';
+import { createSortedTimestamps, createTimestampMap, createFieldName } from './timeseries';
 
-describe('generateSortedTimestamps', () => {
-  test('should compile and sort all timestamps across all series', () => {
-    // 🅐rrange
-    // Points are in the form [timestamp, value]
+beforeAll(() => {
+  // Create a mock template server
+  // nb this is used in the createFieldName tests
+  setTemplateSrv({
+    getVariables() {
+      return [];
+    },
+    replace(target, scopedVars, format) {
+      if (target.includes('$service')) {
+        return target.replace('$service', JSON.stringify(['web', 'android', 'ios']));
+      }
+      return target || '';
+    },
+  });
+});
```

Comment:
This mock template server setup was previously called below, but it's a global setup due to how Grafana `getTemplateSrv` works. I moved it to the start of the test file to make that clearer.